File: src/main/java/kr/teentime/mainApi/repository/MemberRepository.java
package kr.teentime.mainApi.repository;

import kr.teentime.mainApi.domain.Member;
import kr.teentime.mainApi.repository.custom.MemberRepositoryCustom;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MemberRepository  extends JpaRepository<Member, Long>, MemberRepositoryCustom {

    Optional<Member> findByNickName(String nickname);
    Optional<Member> findByPhoneNumber(String phoneNumber);
}

File: src/main/kotlin/kr/teentime/mainApi/domain/member/persistence/repository/MemberRepository.kt
package kr.teentime.mainApi.domain.member.persistence.repository

import kr.teentime.mainApi.domain.member.persistence.entity.MemberEntity
import org.springframework.data.repository.CrudRepository

interface MemberRepository : CrudRepository<MemberEntity, Long> {
    fun findByIdOrNull(memberId: Long): MemberEntity?

}